### PR TITLE
Support Python 3.13

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Related:

- CI blocked by https://github.com/mypyc/mypyc/issues/1056

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
All checks are 🟢 over on my fork: https://github.com/edgarrmondragon/charset_normalizer/actions/runs/10414259271.

That means we may just have to wait for Python 3.13.0rc2 and mypy 1.12 to be released :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -176,8 +183,21 @@ jobs:
...
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.13"
         os: [ ubuntu-latest, macos-latest, windows-latest ]
+        include:
+          - python-version: "3.7"
+            os: ubuntu-latest
+          - python-version: "3.7"
+            os: macos-13
```
Comment (edgarrmondragon) [inline .github/workflows/ci.yml:198]: Python 3.7 isn't available for `macos-latest` (currently `macos-14`) anymore, so I extracted these.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -45,7 +45,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11", "3.12" ]
+        python-version:
+          - "3.7"
```
Comment (louis-jaris) [inline .github/workflows/ci.yml:49]: IMHO you can remove `3.7` that is not supported anymore (it's been a year).

Once you add python 3.13, you could also drop python3.8 (that is official reaching EOL this October, like, in 7 days)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -176,8 +183,21 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11", "3.12" ]
+        python-version:
+          - "3.8"
```
Comment (louis-jaris) [inline .github/workflows/ci.yml:187]: Ditto python 3.8 (end of support starts in 7 days)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (louis-jaris) [reply .github/workflows/ci.yml:198]: IMHO - you can remove python3.7 testing completely given that this is not supported anymore. Additionally, the `README` contains instructions on this lib versions that are still working with dead python versions.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (edgarrmondragon) [reply .github/workflows/ci.yml:49]: Thanks for the suggestion. I think that's up to the maintainer. If the burden of keeping it around is not too big, they might wanna wait a while before dropping it.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Ousret) [reply .github/workflows/ci.yml:198]: We are going to keep Python 3.7 support.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Ousret) [reply .github/workflows/ci.yml:49]: Indeed. No strong reason to remove its support.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Ousret) [reply .github/workflows/ci.yml:187]: Idem.

**Review by hugovk — APPROVED** [review]
Thank you!

**Comment by edgarrmondragon** [comment]
Ok, so https://github.com/python/mypy/issues/17815 seems to be the only blocker 🎉

**Review by Ousret — APPROVED** [review]
LGTM

**Pull request merged**